### DISCUSSION TITLE
Should I create a C program with C++ bindings or a C++ program with C bindings?

### DISCUSSION THREAD

**Discussion by gogo2464** [swig/swig#2834] [Q&A]
Hello,

I could generate C wrapper from C++ see https://github.com/swig/swig/issues/2733 Or I could use the extend: see https://www.swig.org/Doc4.1/SWIG.html#SWIG_adding_member_functions and https://github.com/swig/swig/discussions/2732 to use C code to C++.

In your opinion, wich one is the best way please? The extends way could require more efforts. I am also not sure the C++ will remain 10 years latter.

**Comment by ojwb** [comment]
Duplicate of #2839.

**Reply by gogo2464** [reply]
This new thread is not a duplicate. It focuses on philosophy. not on features.

**Reply by ojwb** [reply]
It still all comes down to the same fundamental practical point - the "C as a target language" branch is not merged and sadly it seems it may never be.
